README.md
# Bash-Shell
Shell scripts Written in Bash


history.md
pgrep Coll
ps -ef | grep Coll
ll -hR /data/rel/
date
ll -hR /data/rel/
ps -ef | grep Coll
ps -ef | grep td-agent
vi /etc/td-agent/conf/td-agent.conf 
vi /etc/td-agent/td-agent.conf
vi /etc/td-agent/conf/td-agent.conf 
vi /etc/td-agent/td-agent.conf
vi /var/log/td-agent/td-agent.log
ll /etc/td-agent/
ll /etc/td-agent/conf/
vi ~/.bash
vi ~/.bash_history 
ll
ll =al
ll -al
find -r name *hisotry
find / name *hisotry
find ~/ name *hisotry
find ~/ -name *hisotry
find ~/ -name .history
find ~/ -name *.history
find ~/ -name *history
find ~/ -name *history*
find / -name *history*
ps -ef | grep td-
ll /etc/td-agent/
vi TrRtMaker.sh 
kafka-console-consumer --zookeeper bd-dev01 --topics tr-rt
kafka-console-consumer --zookeeper bd-dev01 --topics 
ll
vi /data/rel/Tr/tr.json 
vi TrRtMaker.sh 
vi /data/rel/Tr/tr.json 
kafka-
kafka
kafka-topics 
kafka-topics --zookeeper bd-dev01 --list
kafka-topics --zookeeper bd-dev01 --list tr-rt
kafka-console-cunsumer --zookeeper bd-dev01 --list 
kafka-console-cunsumer --zookeeper bd-dev01
kafka-console-consumer --zookeeper bd-dev01 --topics tr-rt
kafka-console-consumer --zookeeper bd-dev01 --topic tr-rt
ll
ll /var/www/html/yum/
find /var/www/html/yum/
find /var/www/html/yum/ -name *lsb*
find /var/www/html/yum/ -name *.tar
find -name /var/www/html/yum/ *lsb*
find -name /var/www/html/yum/ *.tar
find -name / *.tar
find -name / 
find / -name
find 
find --help
ll /var/www/html/yum/
ll /var/www/html/yum/bigdata/
ll /var/www/html/yum/bigdata/td-agent/
ll /var/www/html/yum/bigdata/td-agent/RPMS/
ll /var/www/html/yum/bigdata/RPMS/
ll /var/www/html/yum/bigdata/RPMS/*lsb*
scp /var/www/html/yum/bigdata/RPMS/*lsb* bd-dev02:/root/
kafka-console-consumer --zookeeper bd-dev01 --topics tr-rt
kafka-console-consumer --zookeeper bd-dev01 --topic tr-rt
ll /var/log/td-agent/buffer/
ll /var/log/td-agent/buffer/sise-realtime/
ll /var/log/td-agent/buffer/
ll /var/log/td-agent/
ps -ef | grep Coll
ll /var/log/td-agent/
ll /var/log/td-agent/logs/
ll /var/log/td-agent/logs/daily.log 
date
ll /data/rel/SiseNewsCollector/result/daily/
ps -ef | grep Coll
ps -ef | grep td-agent
ll /etc/td-agent/conf/
ll /var/log/td-agent/
ll /var/log/td-agent/logs/
ll /var/log/td-agent/logs/daily.log 
vi /var/log/td-agent/logs/daily.log 
ps -ef | grep td-agent
pkill td-agent
td-agent &
ps -ef | grep td-agent
ll /var/log/td-agent/
ll /var/log/td-agent/buffer/
ll /etc/td-agent/conf/
ll /etc/td-agent/plugin/
kafka-console-consumer --zookeeper bd-dev01 --topic tr-rt
ll
vi BunbongCollector.sh 
ll
ll /data/rel/
ll /data/rel/Tr/tr
ll /data/rel/Tr/
date
ll /data/rel/Tr/tr_20190703.json 
vi /data/rel/Tr/tr_20190703.json 
ll
ll compress/
ll /etc/td-agent/conf/
vi td-agent.conf 
ll
ll /
ll
mkdir backup_20190621
ll
ll test/
mv td-agent.conf td-agent.conf_20190618 backup_20190621/
mv td-agent.conf backup_20190621/td-agent.conf_20190618
ll
ll backup_20190621/
mv backup_20190621/td-agent.conf backup_20190621/td-agent.conf_20190618
ll
mv openjdk-1.8.0* backup_20190621/
ll
mv jdk-1.8.0-111.tar backup_20190621/
ll jdk1.8.0_setup/
ll
mv fluentd_agents.tar backup_20190621/fluentd_agents.tar_20190621
ll
ll *.conf
ls *.conf
ls *.conf | sed"s/\(.*\)/echo \1/"
ls *.conf | sed"s/\(.*\)/echo \1/" | sh
ls *.conf | sed "s/\(.*\)/echo \1/" | sh
ls *.conf | sed "s/\(.*\)/echo \1_29189618/" | sh
ls *.conf | sed "s/\(.*\)/echo \1 \1_29189618/" | sh
ls *.conf | sed "s/\(.*\)/cp \1 \1_29189618/" | sh
ll
rm daily.conf_29189618 
rm tr1.conf_29189618 
ls *.conf | sed "s/\(.*\)/mv \1 backup_20190621\/\1_20190618/" | sh
ll backup_20190621/
ll
ll compress/
rm -r compress/
ll
ll test/
ll
ll kafka-console 
vi kafka-console 
chmod -x kafka-console 
ll
vi KafkaTopics.sh 
ll
ll make_dir.sh 
vi make_dir.sh 
vi setup.sh 
rm WeblogColletcot.sh 
vi WeblogColletcot.sh 
mv WeblogColletcot.sh WeblogCollector.sh 
ll
mkdir transfer_20190703
cp TrRtMaker.sh transfer_20190703/
cp Collector.sh transfer_20190703/
ll
ll transfer_20190703/
ll
cp /etc/td-agent/conf/td-agent.conf transfer_20190703/
cp /etc/td-agent/conf/tr-rt.conf transfer_20190703/
ll transfer_20190703/
cd transfer_20190703/
ll
vi Collector.sh 
vi TrRtMaker.sh 
vi tr-rt.conf 
vi /var/log/td-agent/logs/tr-rt.log 
vi tr-rt.conf 
ll
vi tr-rt.conf 
ll
cd ../
ll
cd transfer_20190703/
ll
vi bd-dev01_td-agent_setup.sh
vi bd-dev01_td-agent&Collector_setup.sh
jobs
;;
ll
vi bd-dev01_td-agent&Collector_setup.sh
vi bd-dev01_td-agent_and_Collector_setup.sh
ll
jobs
kill %1
kill %2
jobs
fg %1
fg %2
jobs
ll
vi bd-dev01_td-agent_and_Collector_setup.sh 
ls
vi bd-dev01_td-agent_and_Collector_setup.sh 
ll /etc/td-agent/
ll /etc/td-agent/conf/
ll
chown -R /etc/td-agent/
chown -R td-agent:td-agent /etc/td-agent/
ll /etc/td-agent/conf/
ll /etc/td-agent/
ll
chown -R td-agent:td-agent *.conf
ll
vi bd-dev01_td-agent_and_Collector_setup.sh 
ll
cd ../
ll
cd transfer_20190703/
ll
mv bd-dev01_td-agent_and_Collector_setup.sh ../
ll
tar -cf bd-dev01_td-agent_and_Collector_setup.tar *
ll
ll ../
mv ../bd-dev01_td-agent_and_Collector_setup.sh .
ll
pwd
ll /root/transfer_20190703/
ls /root/transfer_20190703/
ll/root
ll /etc/td-agent/td-agent.conf
ll ~/transfer_20190703/
chmod +x bd-dev01_td-agent_and_Collector_setup.sh 
ll
vi bd-dev01_td-agent_and_Collector_setup.
vi bd-dev01_td-agent_and_Collector_setup.sh 
ll
vi bd-dev01_td-agent_and_Collector_setup.sh 
./bd-dev01_td-agent_and_Collector_setup.sh 
ll
ll bd-dev01_td-agent_and_Collector_setup
vi bd-dev01_td-agent_and_Collector_setup
vi bd-dev01_td-agent_and_Collector_setup.sh 
tar -xf bd-dev01_td-agent_and_Collector_setup.tar bd-dev01_td-agent_and_Collector_setup/
l bd-dev01_td-agent_and_Collector_setup/
ll tar -xf bd-dev01_td-agent_and_Collector_setup.tar bd-dev01_td-agent_and_Collector_setup/
ll
ll -hR
ll
ll bd-dev01_td-agent_and_Collector_setup/
tar -xf bd-dev01_td-agent_and_Collector_setup.tar
ll
rm Collector.sh TrRtMaker.sh *.conf
ll /etc/td-agent/
ll /etc/td-agent/conf/
ll
tar -xf bd-dev01_td-agent_and_Collector_setup.tar ./bd-dev01_td-agent_and_Collector_setup/
ll

Part2_OpenJDK.sh
#!/bin/bash

######################Global Variable############################
Version=8
mini_Version=212
backup=/backup
mkdir $backup
#################################################################


echo "=========================================================="
echo "#2-1. Upload "openjdk$Version_$mini_Version.tar.gz""
        echo "
                **********************************************************
                WARNING!!!! DID YOU Upload openjdk$Version_$mini_Version.tar.gz?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
             "
        read
echo "=========================================================="

echo "=========================================================="
echo "#2-2. Uncompress "openjdk$Version_$mini_Version.tar.gz""
tar -xvzf openjdk$Version\_$mini_Version.tar.gz
echo "=========================================================="

#echo "=========================================================="
echo "#2-3. Change Directory to Openjdk Repository"
cd openjdk$Version\_$mini_Version
echo "=========================================================="

echo "=========================================================="
echo "#2-4. install openjdk$Version_$mini_Version"
rpm -Uvh --replacepkgs *rpm
echo "=========================================================="

echo "=========================================================="
echo "#2-5. Check Install status"
yum list installed | grep jdk
echo "=========================================================="

echo "=========================================================="
echo "#2-6. Change JAVA version"
alternatives --config java
echo "=========================================================="

echo "=========================================================="
echo "Check JAVA version"
java -version
echo "=========================================================="


Part3_JVMversion.sh
#!/bin/bash

echo "	  *******************************************************
	go to URL MUST BE CHANGE when scripts run on dev, real server
		laptop is hostname -i Result is 10.0.1.15 172.22.5.101
		dev, real is 172.22.5.101
	  *******************************************************"

######################Global Variable############################
Version=8
mini_Version=212
backup=/backup
mkdir -p $backup
#################################################################

##########################Part 3#################################

if [ -f /etc/default/cloudera-scm-server ]; then
echo "`ls -l /etc/default/cloudera-scm-server`"

echo -e "\n\npart3 ClouderaJDK to OpenJDK\n\n"
echo "=========================================================="
echo "#3-1. Set to JAVA_HOME (Cloudera Manager installed server)"
	#cp /etc/default/cloudera-scm-server /etc/default/cloudera-scm-server_`date +%F`_BAK
	#echo "export JAVA_HOME="/usr/lib/jvm/java-1.8.0-openjdk"" >> /etc/default/cloudera-scm-server_test
	#cat /etc/default/cloudera-scm-server >> /etc/default/cloudera-scm-server_test
	#systemctl restart cloudera-scm-server

	echo "export JAVA_HOME="/usr/lib/jvm/java-1.8.0-openjdk"" >> /etc/default/cloudera-scm-server
	echo "=========================================================="

	echo "=========================================================="
	echo "#3-2. Restart Cloudera Manager Server"
	systemctl restart cloudera-scm-server
	echo "=========================================================="

	echo "=========================================================="
	echo "#3-3. Check OpenJDK Applied to Cloudera Manager Server"
        if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
                echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/help/about"
        else
                echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/help/about"
        fi
	echo "check Java VM Name is OpenJDK 64-Bit Server VM"
	echo "check Java Version is 1.8.0_$mini_Version"
	echo "=========================================================="

else
	echo "=========================================================="
	echo " This server is not cloudera-scm-server installed"
	echo "`ls -l /etc/default/cloudera-scm-server`"
	echo "=========================================================="

fi


Part4_ChangeJVMdir.sh
#!/bin/bash

echo "	  *******************************************************
	go to URL MUST BE CHANGE when scripts run on dev, real server
	  *******************************************************"


######################Global Variable############################
Version=8
mini_Version=212
backup=/backup
mkdir -p $backup
#################################################################

if [ -f /etc/default/cloudera-scm-server ]; then
echo "`ls -l /etc/default/cloudera-scm-server`"

	##########################Part 4#################################

	echo -e "\n\npart4 Config adjust by applying OpenJDK8\n\n"
	echo "=========================================================="
	echo "#4-1 JAVA_HOME direct change to all hosts"
        if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
                curl -X PUT -H "Content-Type: application/json" -u "admin:admin" -d '{  "items": [ { "name" : "java_home", "value" : "/usr/lib/jvm/java-1.8.0-openjdk", "sensitive" : false } ] }' "http://`hostname -I | awk '{print $2}'`:7180/api/v14/cm/allHosts/config"
        else
                curl -X PUT -H "Content-Type: application/json" -u "admin:admin" -d '{  "items": [ { "name" : "java_home", "value" : "/usr/lib/jvm/java-1.8.0-openjdk", "sensitive" : false } ] }' "http://`hostname -I | awk '{print $2}'`:7180/api/v14/cm/allHosts/config"
        fi
	echo "=========================================================="

	echo "=========================================================="
	echo "#4-2 Additional configuration change"
	echo "Just Change & Save. Do NOT RESTART YET"
	echo "=========================================================="

else
        echo "=========================================================="
        echo " This server is not cloudera-scm-server installed"
		echo "`ls -l /etc/default/cloudera-scm-server`"
        echo "=========================================================="


fi


Part1_LocalRepository.sh
#!/bin/bash

######################Global Variable############################
#:<<'COMMENT' / COMMENT is Bash Block Comment#
: <<'COMMENT' 
COMMENT

Old_version=5.9.0

tar_version=62
version=6.2
mini_version=.0

repoDir="/etc/yum.repos.d"
repoFile="$repoDir/cloudera-cdh$tar_version.repo" 
repoFile_cm="$repoDir/cloudera-manager$tar_version.repo"
#################################################################


####################### Part 1 #################################

echo "#Part1 Repository Build"

if [ -f /etc/default/cloudera-scm-server ]; then
	echo "=========================================================="
	echo "#1-1. Upload "cloudera_$tar_version.tar.gz""
	echo "Upload Setup files to server (by. FTP or portable media)"
        echo "
                **********************************************************
                WARNING!!!! DID YOU Upload "cloudera_$tar_version.tar.gz"?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
             "
        read
	echo "=========================================================="

	echo "=========================================================="
	echo "#1-2. Uncompress "cloudera_$tar_version.tar.gz""
	tar -xvzf cloudera_$tar_version.tar.gz
	echo "=========================================================="

	echo "=========================================================="
	echo "#1-3. Change Directory name as repos"
	mv cloudera-repos cloudera-$version
	echo "=========================================================="

	echo "=========================================================="
	echo "#1-4. Copy install files"
	##for Dev_server
	if [ ! -d /opt/BigData/repo ]; then 
		mkdir -p /opt/BigData/repo
	fi
	
	if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
		mv cloudera-$version /opt/BigData/repo
		ln -s /opt/BigData/repo /var/www/html/yum/repo
	else
		cp -R cloudera-$version /opt/BigData/repo
		ln -s /opt/BigData/repo /var/www/html/repo
	fi

	##개발환경이랑 확인필요
	#for Local LapTop
	cd /var/www/html/yum
	#ln -s /opt/cloudera-$version cloudera-$version
	echo "=========================================================="
fi

##개발환경이랑 확인필요
echo "=========================================================="
echo "#1-5. Disable cloudera $Old_version Repositoy config file"
mv $repoDir/c* $repoDir/backup/
yum clean all
yum repolist
echo "=========================================================="

echo "=========================================================="
echo "#1-6. Change Repositoy config as cloudera $version"

#for Local LapTop
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo [cloudera-cdh$version] > $repoFile
	echo name = Cloudera CDH$version, Version >> $repoFile
	echo baseurl = http://172.22.5.101/yum/repo/cloudera-$version/cdh6/$version$mini_version/redhat7/yum >> $repoFile
	echo gpgcheck = 0 >> $repoFile

	echo [cloudera-manager$version] > $repoFile_cm
	echo name = Cloudera Manager, Version $version$mini_version >> $repoFile_cm
	echo baseurl = http://172.22.5.101/yum/repo/cloudera-$version/cm6/$version$mini_version/redhat7/yum >> $repoFile_cm
	echo gpgcheck = 0  >> $repoFile_cm

fi
###for Real_server
if [ `hostname` = "bd-mgr01.local" ]; then

	echo [cloudera-cdh$version] > $repoFile
	echo name = Cloudera CDH$version, Version >> $repoFile
	echo baseurl = http://bd-mgr01.local:10081/repo/cloudera-$version/cdh6/$version$mini_version/redhat7/yum >> $repoFile
	echo gpgcheck = 0 >> $repoFile

	echo [cloudera-manager$version] > $repoFile_cm
	echo name = Cloudera Manager, Version $version$mini_version >> $repoFile_cm
	echo baseurl = http://bd-mgr01.local:10081/repo/cloudera-$version/cm6/$version$mini_version/redhat7/yum >> $repofile_cm
	echo gpgcheck = 0 >> $repofile_cm
fi

##for Dev_server
if [ `hostname` = "bd-dev03.local" ]; then
	echo [cloudera-cdh$version] > $repoFile
	echo name = Cloudera CDH$version, Version >> $repoFile
	echo baseurl = http://bd-dev03.local:10081/repo/cloudera-$version/cdh6/$version$mini_version/redhat7/yum >> $repoFile
	echo gpgcheck = 0 >> $repoFile

	echo [cloudera-manager$version] > $repoFile_cm
	echo name = Cloudera Manager, Version $version$mini_version >> $repoFile_cm
	echo baseurl = http://bd-dev03.local:10081/repo/cloudera-$version/cm6/$version$mini_version/redhat7/yum >> $repoFile_cm
	echo gpgcheck = 0  >> $repoFile_cm

fi

yum clean all
yum repolist
echo "=========================================================="


Part6_UpgradeCM6agent.sh
#!/bin/bash

echo "	  *******************************************************
	go to URL MUST BE CHANGE when scripts run on dev, real server
	  *******************************************************"


######################Global Variable############################
Version=8
mini_Version=212
backup=/backup
mkdir -p $backup
#################################################################

if [ ! -f /etc/default/cloudera-scm-server ]; then
echo "`ls -l /etc/default/cloudera-scm-server`"

	##########################Part 6#################################

	echo -e "\n\n\part6 Cloudera Manger Agent Upgrade to Non Manager server\n\n"
	echo "=========================================================="
	echo "#6-1 Cloudera Manager Agent Stop"
	systemctl stop cloudera-scm-agent
	echo "=========================================================="

	echo "=========================================================="
	echo "#6-2 Upgrade the Packages"
	yum clean all
	yum repolist
	yum -y upgrade cloudera-manager-daemons cloudera-manager-agent
	echo "=========================================================="
		
	echo "=========================================================="
	echo "#6-3 Packages installation"
	rpm -qa 'cloudera-manager-*'
	echo "=========================================================="

	echo "=========================================================="
	echo "#6-4 Cloudera Manager agent Start"
	systemctl start cloudera-scm-agent
	echo "=========================================================="

	echo "=========================================================="
	echo "6-5 Check Cloudera Manager Upgrade"
	if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
		echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/upgrade"
	else
		echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/upgrade"
	fi

	echo "
			**********************************************************
			WARNING!!!! DO YOU FINISH THIS STEP?
			then press AnyKey
			IF Want to stop this Step Press CTRL+C
			**********************************************************
	     "
	read
	echo "=========================================================="

else
        echo "=========================================================="
        echo " This server is cloudera-scm-server installed"
		echo "`ls -l /etc/default/cloudera-scm-server`"
        echo "=========================================================="

fi


Part7_UpgradeCDH6.sh
#!/bin/bash

echo "	  *******************************************************
	go to URL MUST BE CHANGE when scripts run on dev, real server
	  *******************************************************"
######################Global Variable############################
Version=8
mini_Version=212
backup=/backup
mkdir -p $backup
#################################################################


##########################Part 7#################################
echo -e "\n\nPart 7\n\n"
echo "=========================================================="
echo "#7-1 Yarn NodeManager Service releas & reassignment"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/services/21/instances"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/services/21/instances"
fi
echo "every \'NodeManager\' checkbox check & service (releas & reassignment)"
echo "DO NOT START \'NodeManager\'!"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/services/30/config#filterdisplayGroup=%EA%B3%A0%EA%B8%89"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/services/30/config#filterdisplayGroup=%EA%B3%A0%EA%B8%89"
fi
echo -e "change value \'kafka.properties에 대한 Kafka Broker 고급 구성 스니펫(안전 밸브)kafka.properties에 대한 Kafka Broker 고급 구성 스니펫(안전 밸브)\' \nas \ninter.broker.protocol.version = 0.9.0 \nlog.message.format.version = 0.9.0"

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-2 Kafka Version Configure"

echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/services/30/config#filterdisplayGroup=%EA%B3%A0%EA%B8%89"

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-3 other configuration check & adjust"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/"
fi

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read

echo "=========================================================="

echo "=========================================================="
echo "#7-4 CDH Stop"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/"
fi
echo "CDH STOP"

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-5 Install CDH Packages"
yum clean all
yum repolist
yum -y remove hadoop-0.20\* hue-\* crunch llama mahout sqoop2 whirr sqoop2-client

if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	tar -xvzf cdh62_dependency.tar.gz
	cd cdh62_dependency
	rpm -Uvh --replacepkgs *rpm
else
	yum -y install libtidy lsb 
fi
yum -y install avro-tools flume-ng hadoop-hdfs-fuse hadoop-hdfs-nfs3 hadoop-httpfs hadoop-kms hbase-solr hive-hbase hive-webhcat hue impala impala-shell kafka kite kudu oozie pig search sentry sentry-hdfs-plugin solr-crunch solr-mapreduce spark-core spark-python sqoop zookeeper parquet hbase solr

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-6 default JAVA directory Change"
mv /usr/java /usr/java_back

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-7 Cloudera MAnager Agent Restart"
systemctl restart cloudera-scm-agent

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-8 (\*Laptop) Oozie Password Change"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	grant all privileges on oozie.* to 'oozie'@'%' identified by 'oozie'
	flush privileges
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/services/20/config#filterdisplayGroup=%EB%8D%B0%EC%9D%B4%ED%84%B0%EB%B2%A0%EC%9D%B4%EC%8A%A4"
fi
echo "create Oozie password"

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-9 Run the Upgrade CDH Wizard"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/clusters/1/upgradePopup"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/clusters/1/upgradePopup"
fi

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-10 Start Yarn NodeManager "
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/services/21/instances"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/services/21/instances"
fi
echo "every \'NodeManager\' checkbox check & start"

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-11 other configuration check & adjust"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/"
fi

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-12 default port change & kafka version adjust"

if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/config2?task=ALL_PORT_CONFIGURATIONS"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/config2?task=ALL_PORT_CONFIGURATIONS"
fi
echo "HBase Master 웹 UI 포트 : 16010 -> 60010"
echo "HBase RegionServer 웹 UI 포트 : 16030 -> 60030"
echo "DataNode HTTP 웹 UI 포트 : 9864 -> 50075"
echo "NameNode 웹 UI 포트 : 9870 -> 50070"
echo "SAVE"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/services/30/config#filterdisplayGroup=%EA%B3%A0%EA%B8%89"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/services/30/config#filterdisplayGroup=%EA%B3%A0%EA%B8%89"
fi

echo -e "delete value \'kafka.properties에 대한 Kafka Broker 고급 구성 스니펫(안전 밸브)kafka.properties에 대한 Kafka Broker 고급 구성 스니펫(안전 밸브)\' \n as \ninter.broker.protocol.version = 0.9.0 \nlog.message.format.version = 0.9.0"

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-13 HDFS metadata Upgrade"
if [ `free -m | grep ^Mem | awk '{print $2}'` -lt 128000 ]; then
	echo "go to http://`hostname -I | awk '{print $2}'`:7180/cmf/services/22/instances/26/do?command=HdfsFinalizeUpgrade"
else
	echo "go to http://`hostname -I | awk '{print $1}'`:7180/cmf/services/22/instances/26/do?command=HdfsFinalizeUpgrade"
fi

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="

echo "=========================================================="
echo "#7-14 DataBase Backup"
echo "mysqldump -u root -p --all-databases > $backup/alldatabases_6.2_`date +%F`.sql"

echo "
                **********************************************************
                WARNING!!!! DO YOU FINISH THIS STEP?
                then press AnyKey
                IF Want to stop this Step Press CTRL+C
                **********************************************************
     "
read
echo "=========================================================="
